lib/transport.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+const { createRequire } = require('module')
+const caller = require('get-caller-file')
+const { isAbsolute } = require('path')
+
+const ThreadStream = require('thread-stream')
+
+function transport (filename, workerData) {
+  if (!(isAbsolute(filename) || filename.indexOf('file://') === 0)) {
+    const callerFile = caller()
+    const callerRequire = createRequire(callerFile)
+    filename = callerRequire.resolve(filename)
+  }
+
+  const stream = new ThreadStream({
+    filename,
+    workerData,
+    sync: true // TODO should this be configurable?
+  })
+
+  stream.on('ready', function () {
+    stream.unref()
+  })
+
+  return stream
```

Comment:
can we support returning an async function as well a stream, the function takes an a log line object and is called for each log line object

```js
return async (parsedLogline) {
  // do something with the log line 
}
```

Reply:
I don't think this should be supported at this level. However I wrote https://github.com/pinojs/pino-abstract-transport which handles all the cases.